packages/dialtone-mcp-server/src/tools/icons.ts

```diff
@@ -0,0 +1,107 @@
+  const words = normalized.split(/\s+/).filter(w => w.length > 0);
+
+  // Create regex for each word
+  const regexArray = words.map((word: string) => {
+    const escaped = word.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    return new RegExp(escaped, 'i');
+  });
+
+  console.error(`[ICON SEARCH DEBUG] Words:`, words);
+
+  const results: SearchResult[] = [];
+
+  // Iterate through all categories and icons
+  for (const [categoryName, icons] of Object.entries(data.categories)) {
+    for (const [iconName, keywords] of Object.entries(icons)) {
+      // Gather searchable text: icon name + category + keywords
+      const searchableTexts = [
+        iconName.toLowerCase(),
+        categoryName.toLowerCase(),
+        ...keywords.map(k => k.toLowerCase())
+      ];
+
+      const combinedText = searchableTexts.join(' ');
+
+      // Check if ALL regexes match somewhere (AND logic)
+      const allWordsMatch = regexArray.every((regex: RegExp) => regex.test(combinedText));
+
+      if (allWordsMatch) {
+        results.push({
+          type: 'icon',
+          name: iconName,
+          details: {
+            category: categoryName,
+            keywords: keywords
+          },
+          metadata: null  // Icons don't have metadata
+        });
+      }
+    }
+  }
+
+  console.error(`[ICON SEARCH DEBUG] Found ${results.length} matches\n`);
+
+  return { results, notes: [] };
+}
+
+/**
+ * Format icon search results
+ */
+export function formatIconResults(results: SearchResult[], query: string): string {
+  if (results.length === 0) {
+    return `No icons found for "${query}".\n\nTry searching with:\n- Icon name (e.g., "bell", "arrow", "calendar")\n- Category (e.g., "alerts", "communication", "time")\n- Keyword (e.g., "notification", "email", "warning")`;
+  }
+
+  let output = `Found ${results.length} icon${results.length > 1 ? 's' : ''} for "${query}":\n\n`;
+
+  // Group by category for better organization
+  const byCategory: { [category: string]: SearchResult[] } = {};
+  results.forEach((result: SearchResult) => {
+    const cat = result.details.category;
+    if (!byCategory[cat]) {
+      byCategory[cat] = [];
+    }
+    byCategory[cat].push(result);
+  });
+
+  // Show results grouped by category
+  Object.entries(byCategory).forEach(([category, icons]) => {
+    output += `**${category}:**\n`;
+    icons.forEach((icon: SearchResult) => {
+      output += `  • **${icon.name}**`;
+      if (icon.details.keywords && icon.details.keywords.length > 0) {
+        output += ` - ${icon.details.keywords.join(', ')}`;
+      }
+      output += `\n`;
+    });
+    output += `\n`;
+  });
+
+  // Usage example
+  output += `**Usage:**\n`;
+  output += `\`\`\`vue\n`;
+  output += `<template>\n`;
+  output += `  <dt-icon name="${results[0].name}" />\n`;
+  output += `</template>\n\n`;
+  output += `<script>\n`;
+  output += `import { DtIcon } from '@dialpad/dialtone-vue'\n`;
+  output += `</script>\n`;
+  output += `\`\`\`\n`;
```

Comment:
This is not the usage we want to promote, We should be using the tree shakable syntax. The dt-icon component is basically deprecated at this point

<img width="445" height="222" alt="Image" src="https://github.com/user-attachments/assets/91947072-7c18-4da0-8bec-0dc5dc8d5ec5" />